File: src/main/scala/org/lolhens/minechanics/client/render/r2d/TextureFile.scala
package org.lolhens.minechanics.client.render.r2d

import net.minecraftforge.fml.client.FMLClientHandler
import net.minecraft.util.ResourceLocation
import org.lwjgl.opengl.GL11

/**
 * Created by LolHens on 18.09.2014.
 */
class TextureFile(resourceLocation: ResourceLocation) {
  def bind = {
    GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F)
    FMLClientHandler.instance.getClient.getTextureManager.bindTexture(resourceLocation)
  }
}

File: src/main/java/org/lolhens/minechanics/core/client/render/TextureFile.java
package org.lolhens.minechanics.core.client.render;

import net.minecraft.util.ResourceLocation;

import org.lwjgl.opengl.GL11;

import cpw.mods.fml.client.FMLClientHandler;

public class TextureFile {
    private ResourceLocation resourceLocation;

    public TextureFile(ResourceLocation resourceLocation) {
        this.resourceLocation = resourceLocation;
    }

    public void bind() {
        GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
        FMLClientHandler.instance().getClient().getTextureManager().bindTexture(resourceLocation);
    }
}
